packages/table/src/table.tsx

```diff
@@ -432,13 +449,20 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
 
     private columnHeaderCellRenderer = (columnIndex: number) => {
         const props = this.getColumnProps(columnIndex);
+        let loading: boolean;
+        const hasLoadingOption = this.hasLoadingOption(props.loadingOptions, ColumnLoadingOption.HEADER);
+        if (hasLoadingOption) {
+            loading = true;
+        } else if (hasLoadingOption === false) {
+            loading = false;
+        }
```

Comment:
still convinced you can do `const loading = this.hasLoadingOption(props.loadingOptions, ColumnLoadingOption.HEADER);` and it will be either `true`, `false`, or `undefined`, exactly as it is now.

have you tried it?